operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -37,86 +43,118 @@ func ExtractSecret(s *corev1.Secret, secretType lokiv1.ObjectStorageSecretType)
 	return &storageOpts, nil
 }
 
-func extractAzureConfigSecret(s *corev1.Secret) (*storage.AzureStorageConfig, error) {
+func extractAzureConfigSecret(s *corev1.Secret) (*storage.AzureStorageConfig, string, error) {
 	// Extract and validate mandatory fields
 	env := s.Data[storage.KeyAzureEnvironmentName]
 	if len(env) == 0 {
-		return nil, kverrors.New("missing secret field", "field", storage.KeyAzureEnvironmentName)
+		return nil, "", kverrors.New("missing secret field", "field", storage.KeyAzureEnvironmentName)
 	}
 	container := s.Data[storage.KeyAzureStorageContainerName]
 	if len(container) == 0 {
-		return nil, kverrors.New("missing secret field", "field", storage.KeyAzureStorageContainerName)
+		return nil, "", kverrors.New("missing secret field", "field", storage.KeyAzureStorageContainerName)
 	}
 	name := s.Data[storage.KeyAzureStorageAccountName]
 	if len(name) == 0 {
-		return nil, kverrors.New("missing secret field", "field", storage.KeyAzureStorageAccountName)
+		return nil, "", kverrors.New("missing secret field", "field", storage.KeyAzureStorageAccountName)
 	}
 	key := s.Data[storage.KeyAzureStorageAccountKey]
 	if len(key) == 0 {
-		return nil, kverrors.New("missing secret field", "field", storage.KeyAzureStorageAccountKey)
+		return nil, "", kverrors.New("missing secret field", "field", storage.KeyAzureStorageAccountKey)
 	}
 
 	// Extract and validate optional fields
 	endpointSuffix := s.Data[storage.KeyAzureStorageEndpointSuffix]
 
+	contents := bytes.Join([][]byte{env, container, name, key, endpointSuffix}, hashSeparator)
+
+	h := sha1.New()
+	_, err := h.Write([]byte(contents))
+	if err != nil {
+		return nil, "", kverrors.Wrap(err, "failed hashing azure secret contents")
+	}
+	sha1C := fmt.Sprintf("%x", h.Sum(nil))
```

Comment:
I think we can get away with not doing the hash calculation specially for each secret type but just hash all the data contents of the secret ([code suggestion](https://github.com/xperimental/loki/commit/9173a7d52ab9ce2709d14ca59c23be805afc4dbf)).

Reply:
Will apply that suggestion! Looks really nice